comm.go

```diff
@@ -131,12 +140,11 @@ func (p *PubSub) handleNewPeerWithBackoff(ctx context.Context, pid peer.ID, back
 	}
 }
 
-func (p *PubSub) handlePeerEOF(ctx context.Context, s network.Stream) {
+func (p *PubSub) handlePeerEOF(_ context.Context, s network.Stream) {
 	pid := s.Conn().RemotePeer()
-	r := protoio.NewDelimitedReader(s, p.maxMessageSize)
-	rpc := new(RPC)
+	r := msgio.NewVarintReaderSize(s, p.maxMessageSize)
 	for {
-		err := r.ReadMsg(&rpc.RPC)
+		_, err := r.ReadMsg()
```

Comment:
we are leaking from the pool here, lets immediately release.

Reply:
Also note that it is probably a bug (or worse, a malicious peer?) in the other side if they write in their input stream.

I think we should not use a buffer, just read a varint, and if something comes in just kill the peer.

But lets not do this in this pr, we might want to discuss this in an issue and also bring it up in specs as the dual stream requirement recently came up.

Reply:
Agree that we should not use buffer or release it here.

Reply:
> But lets not do this in this pr, we might want to discuss this in an issue and also bring it up in specs as the dual stream requirement recently came up.

The dual stream is definitely something that needs to be "speced out" and I saw a PR from ChainSafe guy to fix this. 

Also, do we need to spec out the way we handle dead peers? It seems like this can vary depending on the implementation, in Go pubsub we only check if the peer is dead by awaiting `EOF` on the stream we write to, but in other implementations, there might be another way to do so. 

Or do you say that specifically, the fact of _reading_ out of _write stream_ has to be handled in a special way and this needs to be discussed and put into specs? Should we make an issue in this repo first?

And another question. Should I just remove/revert the changes from `handleEOF` from the PR so we can merge it?

Reply:
cc @vyzo